recOrder/compute/fluorescence_compute.py

```diff
@@ -0,0 +1,148 @@
+
+
+    print('Initializing Reconstructor...')
+    start_time = time.time()
+
+    reconstructor = fluorescence_microscopy(img_dim=img_dim,
+                                            lambda_emiss=wavelength_nm/1000,
+                                            ps=pixel_size_um/magnification,
+                                            psz=z_step_um,
+                                            NA_obj=NA_obj,
+                                            n_media=n_obj_media,
+                                            deconv_mode=deconv_mode,
+                                            pad_z=pad_z,
+                                            use_gpu=use_gpu,
+                                            gpu_id=gpu_id)
+
+    elapsed_time = (time.time() - start_time) / 60
+    print(f'Finished Initializing Reconstructor ({elapsed_time:0.2f} min)')
+
+    return reconstructor
+
+
+#TODO: figure out robust background correction method
+def calculate_background(data):
+    """
+    Calculate the background with uni-modal thresholding
+
+    Parameters
+    ----------
+    data:           (nd-array) data array of size (N_fluor, Y, X)
+
+    Returns
+    -------
+    background_vals:    (list) list of average background values of size [N_fluor]
+    """
+
+    if data.ndim == 3:
+        fluors = data.shape[0]
+    elif data.ndim == 2:
+        fluors = 1
+        data = data[np.newaxis, :, :]
+    else:
+        raise ValueError('invalid input data dimensions.  Data must be (N_fluor, Y, X) or (Y, X)')
+
+    background_vals = []
+    for i in range(fluors):
+        background_vals.append(get_unimodal_threshold(data[i]))
+
+    return background_vals
+
+
+def deconvolve_fluorescence_2D(data, reconstructor: fluorescence_microscopy, bg_level, reg=1e-4):
+    """
+    Deconvolve 2D fluorescence image(s).  Will loop through multiple fluorescence channels
+
+    Parameters
+    ----------
+    data:               (nd-array) 2D data of size (N_fluor, Y, X) or (Y, X)
+    reconstructor:      (waveorder.fluorescence_microscopy) Initialized reconstructor object
+    bg_level:           (list or int) list/int of background levels of size [N_Fluor] or 1
+    reg:                (list or int) list/int of regularization parameters of size [N_Fluor] or 1
+
+    Returns
+    -------
+    deconvolved_data:   (nd-array) of size (N_fluor, Y, X) or (Y, X)
+
+    """
+
+    if data.ndim > 3:
+        raise ValueError('invalid input data dimensions.  Data must be (N_fluor, Z, Y, X) or (Z, Y, X)')
+
+    deconvolved_data = reconstructor.deconvolve_fluor_2D(data, bg_level, reg)
+
+    return deconvolved_data
+
+
+def deconvolve_fluorescence_3D(data, reconstructor: fluorescence_microscopy, bg_level, reg=1e-4):
+    """
+    Deconvolve 3D fluorescence volume(s).  Will loop through multiple fluorescence channels
+
+    Parameters
+    ----------
+    data:               (nd-array) 2D data of size (N_fluor, Z, Y, X) or (Z, Y, X)
```

Comment:
Minor: `reconstruct_phase3D` function receives an array with dimension (Y, X, Z) and `deconv_fluorescence_3D` receives an array with dimension (Z, Y, X). We can either make the input of the `reconstruct_phase3D` as (Z, Y, X) or stay what it is and wait for an update in `waveorder`. It is not a big priority, but I note it as an action needed sometime later.

Reply:
Ah thanks for catching this -- I think we can leave this for now as it has proper documentation.  It will be nice if `waveorder` can conform to more of the OME standardization of arrays being (T, C, Z, Y, X).